src/Voron/Impl/Paging/CryptoPager.cs

```diff
@@ -114,12 +112,26 @@ private static int GetNumberOfPages(PageHeader* header)
             return checked((overflowSize / Constants.Storage.PageSize) + (overflowSize % Constants.Storage.PageSize == 0 ? 0 : 1));
         }
 
-        public override byte* AcquirePagePointerForNewPage(IPagerLevelTransactionState tx, long pageNumber, int numberOfPages, PagerState pagerState = null)
+        public override byte* AcquirePagePointerForNewPage(IPagerLevelTransactionState tx, long pageNumber, int numberOfPages, PagerState pagerState = null, bool handleOverwrite = false)
```

Comment:
Under what scenario we want the current behavior of `handleOverwrite = false`?

Reply:
The second usage of `AcquirePagePointerForNewPage` is when we allocate the scratch page. However, I think you're right, in that scenario the behavior `handleOverwrite = true` should also be active.

I'll drop `handleOverwrite` optimal parameter and make it as the default behavior